src/algorithm/distance.rs

```diff
@@ -27,10 +67,235 @@ impl<T> Distance<T, Point<T>> for Point<T>
     }
 }
 
+// Return minimum distance between a Point and a Line segment
+// This is a helper for Point-to-LineString and Point-to-Polygon distance
+// adapted from http://stackoverflow.com/a/1501725/416626
+fn line_segment_distance<T>(point: &Point<T>, start: &Point<T>, end: &Point<T>) -> T
+    where T: Float + ToPrimitive
+{
+    let dist_squared = pow(start.distance(end), 2);
+    // Implies that start == end
+    if dist_squared == T::zero() {
+        return pow(point.distance(start), 2);
+    }
+    // Consider the line extending the segment, parameterized as start + t (end - start)
+    // We find the projection of the point onto the line
+    // This falls where t = [(point - start) . (end - start)] / |end - start|^2, where . is the dot product
+    // We clamp t from [0.0, 1.0] to handle points outside the segment start, end
+    let t = T::zero().max(T::one().min((*point - *start).dot(&(*end - *start)) / dist_squared));
+    let projected = Point::new(start.x() + t * (end.x() - start.x()),
+                               start.y() + t * (end.y() - start.y()));
+    point.distance(&projected)
+}
+
+#[derive(PartialEq, Debug)]
+struct Mindist<T>
+    where T: Float
+{
+    distance: T,
+}
+// These impls give us a min-heap when used with BinaryHeap
+impl<T> Ord for Mindist<T>
+    where T: Float
+{
+    fn cmp(&self, other: &Mindist<T>) -> Ordering {
+        other.distance.partial_cmp(&self.distance).unwrap()
+    }
+}
+impl<T> PartialOrd for Mindist<T>
+    where T: Float
+{
+    fn partial_cmp(&self, other: &Mindist<T>) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl<T> Eq for Mindist<T> where T: Float {}
+
+// Minimum distance from a Point to a Polygon
+impl<T> Distance<T, Polygon<T>> for Point<T>
+    where T: Float
+{
+    fn distance(&self, polygon: &Polygon<T>) -> T {
+        // No need to continue if the point is fully inside
+        if polygon.contains(self) { return T::zero() }
+        // minimum priority queue
+        let mut dist_queue: BinaryHeap<Mindist<T>> = BinaryHeap::new();
+        // get exterior ring
+        let exterior = &polygon.0;
+        // exterior ring as a LineString
+        let ext_ring = &exterior.0;
+        for chunk in ext_ring.chunks(2) {
+            let dist = match chunk.len() {
+                2 => line_segment_distance(self, chunk.first().unwrap(), chunk.last().unwrap()),
+                _ => {
+                    // final point in an odd-numbered exterior ring
+                    line_segment_distance(&self, chunk.first().unwrap(), chunk.first().unwrap())
+                }
+            };
```

Comment:
Alternatively:

```rust
line_segment_distance(self, chunk[0], chunk.last().or(chunk[0]));
```

The slice returned by the `chunks` iterator guarantees that element `[0]` exists.